modules/Evse15118D20/charger/ISO15118_chargerImpl.cpp

```diff
@@ -112,15 +177,105 @@ void ISO15118_chargerImpl::ready() {
     };
     const auto callbacks = create_callbacks();
 
+    if (mod->config.supported_dynamic_mode) {
+        setup_config.control_mobility_modes.push_back(
+            {iso15118::message_20::ControlMode::Dynamic, iso15118::message_20::MobilityNeedsMode::ProvidedByEvcc});
+        if (mod->config.supported_mobility_needs_mode_provided_by_secc) {
+            setup_config.control_mobility_modes.push_back(
+                {iso15118::message_20::ControlMode::Dynamic, iso15118::message_20::MobilityNeedsMode::ProvidedBySecc});
+        }
+    }
+
+    if (mod->config.supported_scheduled_mode) {
+        setup_config.control_mobility_modes.push_back(
+            {iso15118::message_20::ControlMode::Scheduled, iso15118::message_20::MobilityNeedsMode::ProvidedByEvcc});
+    }
+
+    if (setup_config.control_mobility_modes.empty()) {
+        EVLOG_warning << "Control mobility modes are empty! Setting dynamic mode as default!";
+        setup_config.control_mobility_modes.push_back(
+            {iso15118::message_20::ControlMode::Dynamic, iso15118::message_20::MobilityNeedsMode::ProvidedByEvcc});
+    }
+
```

Comment:
would write a helper function which does `setup_config.control_mobility_modes = helper_init_from_config(mod->config)` (with an appropriate name of course).

Reply:
Added a helper function